### PR TITLE
Wdp200103 12

### PR DESCRIPTION
**Pull request by mariPab**
The purpose was to add new functionality to choose products to comparison and display them on sticky bar at the bottom of the viewport.
Clicking compare button adds product to comparison array which is stored in Redux store. It is possible to add only 4 unique products. On hover there is close button that removes product from comparison array.
Compare button at the left of sticky bar does nothing.

### CODE REVIEW

**Comment by Varenthein** [comment]
@konradkukier2 Review request.

File: src/redux/comparisonRedux.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export const removeFromCompare = payload => ({ payload, type: REMOVE_FROM_COMPARE });
+
+/* reducer */
+export default function reducer(statePart = [], action = {}) {
+  switch (action.type) {
+    case ADD_TO_COMPARE: {
+      let index = statePart.findIndex(product => product.id === action.payload.id);
+      return statePart.length < 4 && index === -1
+        ? [...statePart, { ...action.payload, id: action.payload.id }]
+        : [...statePart];
```
Comment (konradkukier2) [inline src/redux/comparisonRedux.js]: w tym case nic nie zmieniamy, możemy zwrócić `statePart` (bez spread operatora)

File: src/redux/comparisonRedux.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+    case REMOVE_FROM_COMPARE: {
+      return [...statePart.filter(product => product.id !== action.payload.id)];
```
[the remark this answers is not in the document]
Comment (mariPab) [reply src/redux/comparisonRedux.js]: okej, dzięki! poprawione

File: src/redux/comparisonRedux.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+/* action creators */
+export const addToCompare = payload => ({ payload, type: ADD_TO_COMPARE });
+export const removeFromCompare = payload => ({ payload, type: REMOVE_FROM_COMPARE });
+
+/* reducer */
+export default function reducer(statePart = [], action = {}) {
+  switch (action.type) {
+    case ADD_TO_COMPARE: {
+      let index = statePart.findIndex(product => product.id === action.payload.id);
+      return statePart.length < 4 && index === -1
```
[the remark this answers is not in the document]
Comment (mariPab) [reply src/redux/comparisonRedux.js]: ok, zapisałam to w zmiennej maxAmountToCompare

File: src/components/common/ProductBox/ProductBoxContainer.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+const mapStateToProps = state => ({
+  comparison: getProductsToCompare(state),
+});
+
+const mapDispatchToProps = (dispatch, props) => ({
+  addToCompare: () =>
+    dispatch(
+      addToCompare({
+        id: props.id,
```
[the remark this answers is not in the document]
Comment (mariPab) [reply src/components/common/ProductBox/ProductBoxContainer.js]: racja, skróciłam zapis

File: src/redux/comparisonRedux.js (hunk 1 above)
Comment (mariPab) [reply src/redux/comparisonRedux.js]: ok, poprawione również

**Comment by konradkukier2** [comment]
LGTM

**Pull request merged**